### DISCUSSION TITLE
Use SAF instead of requesting storage permissions

### DISCUSSION THREAD

**Discussion by matchboxbananasynergy** [libre-tube/LibreTube#1130] [Ideas]
I understand that features such as subscription import and video/audio download currently requires storage permissions.

NewPipe allows you to do this without having to grant it any kind of storage permissions by bringing up the file manager and having you choose a directory that you want the download to go to.

Is there a technical reason why Libretube can't do away with this invasive permission?

I'm excited to see how the app shapes up, by the way, great work on it so far!

**Comment by Bnyro** [comment]
Yeah, that's what issue #557 is for.

However, I'm not really motivated to work on downloads any further because I already spent a lot of time in making downloads halfway usable which is a bit tricky due to the new storage system since Android 11.

I appreciate and accept any pull request that improves or changes how downloads etc. work, nonetheless I doubt there is anyone who is willing to work on it.

Regards

**Comment by Bnyro** [comment]
That's how it works now with v0.6.0 :)

**Reply by matchboxbananasynergy** [reply]
I'm very glad to see that the storage permissions have been removed! Ideally, it would be nice for downloads to be written to /Downloads, but I think the in-app download feature as you've designed it works very well for now!
